### PR TITLE
Add invoke targets for ruff formatting/linting

### PR DESCRIPTION
**Pull request by induane**
- Removed dependency on black
- Removed dependency on isort
- Fixed a ton of linter complaints
- Updated ruff dependency
- Added invoke target for linter check
- Added invoke target for code formatting
- Updated pre-commit hook to use newer "ruff format" subcommand

### CODE REVIEW

File: observability_api/endpoints/v1/rules.py [hunk 1]
```diff
@@ -31,37 +31,40 @@ def _validate_rule_action(rule: Rule) -> None:
...
         ---
         tags: ["Rule"]
         description: List all rules that attached to a journey.
         operationId: ListRules
         security:
           - SAKey: []
-        parameters:
+
+        Parameters
+        ----------
```
Comment (rboni-dk) [inline observability_api/endpoints/v1/rules.py:67]: These docstrings are yaml and this particular change breaks the parser. Can you revert these?

**Review by rboni-dk — CHANGES_REQUESTED** [review]
Looks good. Thanks for the patch!

Left a couple of change requests.

Also, !20 got obsolete, right?

File: observability_api/endpoints/v1/rules.py (hunk 1 above)
Comment (induane) [reply observability_api/endpoints/v1/rules.py:67]: Oh lordy ya'll went with yaml docstrings? I pity thee! :smile:

File: common/entities/component_meta.py [hunk 2]
```diff
@@ -18,11 +18,11 @@ class SimpleComponentMeta(ModelBase):
     is identified. This value gets stored at the `type` field of the Component entity.
     """
 
-    def __new__(mcs, name: str, bases: tuple[type, ...], attrs: dict[str, object]) -> Type[BaseModel]:
+    def __new__(cls, name: str, bases: tuple[type, ...], attrs: dict[str, object]) -> type[BaseModel]:
```
[the remark this answers is not in the document]
Comment (induane) [reply common/entities/component_meta.py:21]: I suppose I could :grin:  -  it'll hurt me though. SO much! THE PAIN! :wink:

[the remark this answers is not in the document]
Comment (rboni-dk) [reply common/entities/component_meta.py]: Cool! Thanks, @induane !

**Review by rboni-dk — APPROVED** [review]

**Pull request merged**